lib/DataSet.js

```diff
@@ -198,8 +198,8 @@ DataSet.prototype._trigger = function (event, params, senderId) {
  * Add data.
  * Adding an item will fail when there already is an item with the same id.
  * @param {Object | Array} data
- * @param {String} [senderId] Optional sender id
- * @return {Array} addedIds      Array with the ids of the added items
+ * @param {string} [senderId] Optional sender id
+ * @return {Array.<string|number>} addedIds      Array with the ids of the added items
```

Comment:
Where does this syntax for the type come from? I can't see it [here](http://usejsdoc.org/tags-param.html).
Can you supply a link to the reference doc you used? Just want to be sure that it's legal.

'Array with elements that are either string or number` - actually wouldn't know how to do that my my knowledge of jsdoc.

Reply:
The jsdoc type definition here was actually partly generated by my IDE. A quick google brings up https://github.com/jsdoc3/jsdoc/issues/714 which is similar.

It would be my preference to create a `@typedef` for an `Id` type, and compose it of `{string|number}`, and reference that, but I am trying to iterate my way there, rather than do too many changes in one big bang.

Reply:
Yeah. I realized later that I was using an old reference for jsdoc, my bad.
Iterating wrt. `@typedef` is the way to go, agreed.